pennylane/templates/tensornetworks/mps.py

```diff
@@ -0,0 +1,177 @@
+        self.loc = loc
+
+        _n_blocks = int(n_wires/(loc/2)-1)
+
+        if shape[-1] != _n_blocks:
+            raise ValueError(
+                f"Weights tensor must have last dimension of length {_n_blocks}; got {shape[-1]}"
+            )
+        else:
+            self.n_blocks = _n_blocks
+
+        allowed_templates = [qml.templates.BasicEntanglerLayers, qml.templates.StronglyEntanglingLayers] #, qml.templates.layers.SimplifiedTwoDesign]
+
+        if block not in allowed_templates: 
+            raise AssertionError("The input block is not included yet in the allowed templates block. Try use MPS_from_function instead.")
+
+        self.block = block
+
+        if block_arg == None:
+
+            self.block_arg = {}
+
+        else:
+            self.block_arg = block_arg
+            self.block_specified()
+
+
+        self.ind_gates = compute_indices_MPS(n_wires, loc)
+
+        super().__init__(weights, wires=wires, do_queue=do_queue, id=id)
+
+    def block_specified(self):
+
+        # An naive implementation to further specify the template block
+        # Non-essential for now, a function to further specified the function
+
+        if self.block == qml.templates.BasicEntanglerLayers and "rotation" in self.block_arg.keys():
+
+            self.block =lambda weights, wires: qml.templates.BasicEntanglerLayers(weights, wires, rotation=self.block_arg["rotation"])
+
+        if self.block == qml.templates.StronglyEntanglingLayers and  "ranges" in self.block_arg.keys():
+
+            self.block = lambda weights, wires: qml.templates.StronglyEntanglingLayers(weights, wires, ranges=self.block_arg["ranges"]) 
+
+        #if self.block == qml.templates.layers.SimplifiedTwoDesign and "initial_layer_weights" in self.block_arg.keys():
+
+        #    self.block = lambda weights, wires: qml.templates.SimplifiedTwoDesign(initial_layer_weights=self.block_arg["initial_layer_weights"], weights, wires)
+
+        return
+
+        
+    def expand(self):
+        with qml.tape.QuantumTape() as tape:
+            for idx, w in enumerate(self.ind_gates):
+                self.block(weights=self.weights[...,idx], wires=w.tolist()) # MEASURE THE LAST QUBIT
+        return tape
+
+
+    @staticmethod
+    def shape(n_wires, block_shape):
+
+        r"""Returns the expected shape of the weights tensor.
+        Args:
+            n_wires (int): total number of wires
+            block_shape (tuple[int]): shape of the block circuit in (n_layers [int], loc [int], n_params [int])
+        Returns:
+            tuple[int]: shape
+        """
+
+        loc = block_shape[1]
+
+        #assert n_wires%2 == 0, "n_wires should be an even integer"
+        #assert loc%2 == 0 and loc != 0, "loc should be an even integer"
+        #assert loc <= n_wires, "loc should be smaller or equal than num_wires"
+
+        n_blocks = int(n_wires/(loc/2)-1)
+
+        return block_shape + (n_blocks,)
+
+class MPS_from_function(Operation):
+    
```

Comment:
```suggestion
    r"""Generates a circuit that follows the matrix product state (MPS) architecture. 
    The individual tensors in the network are replaced with user provided circuits.
    Args:
        wires (Iterable): The wires that the template acts on. 
        loc (int): The number of wires in a single tensor block. This value must be even.
        block (callable): A function that defines the gates for all tensor blocks.
        n_params_block (int): The number of trainable parameters per block.
        weights (Iterable): The values given to the trainable parameters.
        
        .. UsageDetails::
            An example of how to use this template is shown below:

        .. code-block:: python
            TODO
    """
```
The codefactor warning here means that we didn't include a description for the `MPS_from_function` class.
I wrote part of this, but the example in usage details still needs to be added.